src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobFileStore.cs

```diff
@@ -296,10 +296,25 @@ public async Task<Stream> GetStreamingFileAsync(long version, Partition partitio
         });
     }
 
+    /// <inheritdoc />
+    public async Task<long> GetFilePropertiesContentLengthAsync(long version, Partition partition, FileProperties fileProperties, CancellationToken cancellationToken)
+    {
+        if (fileProperties is not null && fileProperties.ContentLength != 0)
+        {
+            return fileProperties.ContentLength;
+        }
+        FileProperties fp = await GetFilePropertiesAsync(version, partition, fileProperties, cancellationToken);
+        return fp.ContentLength;
+    }
+
     /// <inheritdoc />
     public async Task<FileProperties> GetFilePropertiesAsync(long version, Partition partition, FileProperties fileProperties, CancellationToken cancellationToken)
     {
+        // until we have backfilled contentLength, we have to continue to support retrieving file properties off of 
```

Comment:
How are we going to backfill this entirely so that this logic is no longer necessary? Will we say that all of these preview IDP services are invalid? Will we write a function like we're done in the past to enumerate the data?